### PR TITLE
test/fakeintake: add apm stats

### PR DESCRIPTION
**Pull request by ahmed-mez**
### What does this PR do?

AIT-9437

- Implement APM stats decoder in the fake intake
- Implement APM stats getter in the fake intake client
- Implement the fakeintakectl get apmstats command

### Motivation

Required for trace-agent e2e testing - related to https://github.com/DataDog/datadog-agent/pull/22003

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/fakeintake/aggregator/apmStatsAggregator.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+	"github.com/tinylib/msgp/msgp"
+)
+
+// APMStatsPayload is a payload type for APM stats
+// It implements PayloadItem
+type APMStatsPayload struct {
+	*pb.StatsPayload
+	collectedTime time.Time
+}
```
Comment (L3n41c) [inline test/fakeintake/aggregator/apmStatsAggregator.go:22]: I don’t know the APM payloads well enough. So maybe what I’m about to suggest doesn’t apply for this usecase.
But, would it make sense to have the fake intake client unbundle the messages that have been grouped by the agent in a single message ?

I mean:
```suggestion
type APMStatsPayload struct {
	*pb.ClientStatsPayload
	collectedTime time.Time
}
```

and then… (see next comment)

File: test/fakeintake/aggregator/apmStatsAggregator.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	rc, err := getReadCloserForEncoding(payload.Data, payload.Encoding)
+	if err != nil {
+		return nil, err
+	}
+	defer rc.Close()
+	var p pb.StatsPayload
+	if err := msgp.Decode(rc, &p); err != nil {
+		return nil, err
+	}
+	return []*APMStatsPayload{{StatsPayload: &p, collectedTime: payload.Timestamp}}, nil
```
Comment (L3n41c) [inline test/fakeintake/aggregator/apmStatsAggregator.go:53]:
```suggestion
    stats := make([]*APMStatsPayload, 0, len(p.Stats)
    for _, stat := range p.Stats {
      stats = append(stats, &APMStatsPayload{StatsPayload: &stat, collectedTime: payload.Timestamp})
    }
	return stats, nil
```

This “unbundling” is what is done for the other types of payloads:
* https://github.com/DataDog/datadog-agent/blob/e3084595ca29763ec0784c6de691a1f8c1873a5b/test/fakeintake/aggregator/metricAggregator.go#L55-L60
* https://github.com/DataDog/datadog-agent/blob/e3084595ca29763ec0784c6de691a1f8c1873a5b/test/fakeintake/aggregator/containerLifecycleAggregator.go#L56-L60

**Review by L3n41c — APPROVED** [review]
I only left what is more a question than a suggestion.

File: test/fakeintake/aggregator/apmStatsAggregator.go (hunk 1 above)
Comment (ahmed-mez) [reply test/fakeintake/aggregator/apmStatsAggregator.go:22]: It's indeed an interesting idea but in that case we would miss top level fields like `AgentHostname` `AgentEnv` `AgentVersion` `ClientComputed` and `SplitPayload` which I believe we'll need in the tests later. For instance `ClientComputed` is crucial to know whether the payload was computed by the tracer or the agent and we'll need to verify that field in the tests.

**Review by pducolin — APPROVED** [review]

**Pull request merged**